apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -315,6 +315,57 @@ public async Task Trace_IgnoreHealthChecks()
                 Assert.Null(trace);
             }
         }
+
+        [Fact]
+        public async Task Trace_TimedBuffer_Stress()
+        {
+            var uri = $"/Trace/{_testId}";
+            // Not the best ever stress test but we are limited by read quotas anyway.
+            var requests = 300;
+            IList<Task<HttpResponseMessage>> responseTasks = new List<Task<HttpResponseMessage>>(300);
+            using (var server = new TestServer(new WebHostBuilder().UseStartup<TraceTestTimedBufferHighQpsApplication>()))
+            using (var client = server.CreateClient())
+            {
+                for (int i = 0; i < requests; i++)
+                {
+                    responseTasks.Add(client.GetAsync(uri));
+                }
+
+                await Task.WhenAll(responseTasks);
+            }
+
+            // TODO: When #2834 is fixed, we should always get as many traces as requests.
+            var minTraces = (int) ((9.0 / 10) * requests);
+            var traces = _polling.GetTraces(uri, _startTime, minEntries: minTraces);
+            Assert.InRange(traces.Count(), minTraces, requests);
+        }
+
+        [Fact]
+        public async Task Trace_SizedBuffer_Stress()
+        {
+            var uri = $"/Trace/{_testId}";
+            // Not the best ever stress test but we are limited by read quotas anyway.
+            var requests = 300;
+            IList<Task<HttpResponseMessage>> responseTasks = new List<Task<HttpResponseMessage>>(300);
+            using (var server = new TestServer(new WebHostBuilder().UseStartup<TraceTestBufferHighQpsApplication>()))
+            using (var client = server.CreateClient())
+            {
+                for (int i = 0; i < requests; i++)
+                {
+                    responseTasks.Add(client.GetAsync(uri));
+                }
+
+                await Task.WhenAll(responseTasks);
+            }
+
+            // TODO: When #2834 is fixed, we should always get as many traces as requests.
+            // Althouhg in the case of a sized buffer we might be missing the last couple of traces
```

Comment:
Typos in here to fix if #2834 being fixed doesn't mean we can get rid of the whole comment:

- Althouhg => although
- din't => didn't